source/client/stream_decoder.h

```diff
@@ -78,14 +88,21 @@ class StreamDecoder : public Envoy::Http::StreamDecoder,
     caller_completion_callback_ = std::move(caller_completion_callback);
   }
 
-  void reset() { complete_ = false; }
+  void reset() {
+    complete_ = false;
+    have_first_body_byte_ = false;
+    milestone_tracker_.reset();
+    // TODO(oschaaf): this is kind of a hack to mark this here.
```

Comment:
Can you expand the comment a bit to explain why it's a hack, and what would be better?

Reply:
comment updated